src/js/components/ExportButton.js

```diff
@@ -72,7 +72,11 @@ export default function ExportButton() {
 
   return (
     <div title="Export search results to ZNG file">
-      <ToolbarButton icon={<ExportIcon />} onClick={onClick} />
+      <ToolbarButton
+        icon={<ExportIcon />}
+        onClick={onClick}
+        data-test-locator="export-results"
```

Comment:
@mikesbrown what do you think about simply having a string like this in the data attribute in react. Then in the spectron test we do:

```js
// in the lib file
export const locator = (name: string) =>`[data-test-locator]="${name}"`

// in the test
await click(app, locator("export-results"))

```

Reply:
Was this because you find the `reactElementProps()` and `integration.js` dance tedious? I understand. Let me outline some reasonings for why that's used:

1. We can use static analysis to find places where we make mistakes. Two examples:
   1. Such attribute names need to be unique in the product, and a convenient way to ensure uniqueness is to make them all keys in the same object so that `eslint` can warn you if there are duplicates.
   1. Mistyping `data-test-locator` as a string literal is more prone to errors and typos and won't be caught in linters, flow, etc., whereas calling `reactElementProps()` is going to have more chance for a mistake to be caught in static analysis. For example, if you mistype `reactElemntProps()` then you'll get an import error from `eslint`. If you typo the string literal `data-tets-locator` on the other hand, the app will happily pass all checks, and you'll wonder why your integration test is hanging trying to find the element.
1. Ease-of-maintenance: Tests shouldn't directly interact with locators. Locators are abstracted away so that if they move, change names, etc., fewer test changes have to be made. For example, you'd likely only need change `src/js/test/integration.js` and possibly `itest/lib/app.js`.

With these considerations, I'm happy to have a conversation about how to rework this to try to make this less tedious but also allow errors to be caught earlier rather than later. Let me know what you think.

Reply:
This got brought up again in #822